contrib/endpoints/src/api_manager/method_impl.h

```diff
@@ -175,6 +183,9 @@ class MethodInfoImpl : public MethodInfo {
 
   // Whether the response is streaming or not.
   bool response_streaming_;
+
+  // map of metric and its cost
+  std::map<std::string, int64_t> metric_cost_map_;
```

Comment:
one minor thing, could consider using std::unorderded_map or std::vector<std::pair<std::string, int>>. 
I recommend using vector, since api_manager would not call find, it just iterates the list.